app/validators/schema_validator.rb

```diff
@@ -0,0 +1,74 @@
+require "json-schema"
+
+class SchemaValidator
+  def initialize(type:, schema_name: nil, schema: nil)
+    @type = type
+    @schema = schema
+    @schema_name = schema_name
+  end
+
+  def validate(payload)
+    @payload = payload
+
+    return true if schema_name_exception?
+
+    errors = JSON::Validator.fully_validate(
+      schema,
+      payload,
+      errors_as_objects: true,
+    )
+
+    return true if errors.empty?
+
+    errors = errors.map { |e| present_error(e) }
```

Comment:
I made a tweak to this and the present_error method itself in https://github.com/alphagov/publishing-api/pull/417/commits/270d6b6f8cbce9cd0f010058e9fc1c0bf87b11b5 because Airbrake/Errbit doesn't understand arrays, so shows the list of errors as a hard-to-read string.